sympy/integrals/integrals.py

```diff
@@ -394,6 +395,12 @@ def doit(self, **hints):
             return S.Zero
 
         # now compute and check the function
+        if isinstance(self.function, Sum):
+            for variable in self.variables:
+                if variable in self.function.limits[0]:
```

Comment:
But we are checking if `self`'s variables are in `self.function`'s limits right ?. Where is it going wrong ?